contracts/Staking.sol

```diff
@@ -113,148 +96,189 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
+        _increaseActiveLockAmount(msg.sender, _lockManager, _amount);
     }
 
     /**
-     * @notice Transfer `_amount` tokens to `_to``_toLockId > 0 ? '\'s lock #' + _toLockId : ''`
+     * @notice Transfer `_amount` tokens to `_to``_toLockManager != 0 ? '\'s lock ' + _toLockManager : ''`
      * @param _to Recipient of the tokens
-     * @param _toLockId Lock id of the recipient to add the tokens to, if any
+     * @param _toLockManager Manager of the recipient lock to add the tokens to, if any
      * @param _amount Number of tokens to be transferred
      */
-    function transfer(address _to, uint256 _toLockId, uint256 _amount) external isInitialized {
+    function transfer(address _to, address _toLockManager, uint256 _amount) external isInitialized {
         // have enough unlocked funds
-        require(_amount <= unlockedBalanceOf(msg.sender), ERROR_NOT_ENOUGH_BALANCE);
+        require(_amount <= _unlockedBalanceOf(msg.sender), ERROR_NOT_ENOUGH_BALANCE);
 
-        _transfer(msg.sender, 0, _to, _toLockId, _amount);
+        _transfer(msg.sender, address(0), _to, _toLockManager, _amount);
     }
 
     /**
-     * @notice Transfer `_amount` tokens from `_from`'s lock #`_fromLockId` to `_to``_toLockId > 0 ? '\'s lock #' + _toLockId : ''`
+     * @notice Transfer `_amount` tokens from `_from`'s lock by `msg.sender` to `_to``_toLockManager > 0 ? '\'s lock by ' + _toLockManager : ''`
      * @param _from Owner of locked tokens
-     * @param _fromLockId Id of the lock for the given account
      * @param _to Recipient of the tokens
-     * @param _toLockId Lock id of the recipient to add the tokens to, if any
+     * @param _toLockManager Manager of the recipient lock to add the tokens to, if any
      * @param _amount Number of tokens to be transferred
      */
     function transferFromLock(
         address _from,
-        uint256 _fromLockId,
         address _to,
-        uint256 _toLockId,
+        address _toLockManager,
         uint256 _amount
     )
         external
-        isLockManager(_from, _fromLockId)
+        isInitialized
     {
-        // No need to check that lockId > 0, as isLockManager would fail
-        // No need to check that have enough locked funds, as _updateActiveLockAmount will fail
+        Account storage account = accounts[_from];
+        Lock storage lock = account.locks[msg.sender];
+        // check that lock is enough, it also means that lock.amount > 0 and therefore hasn't been unlocked
+        require(lock.amount >= _amount, ERROR_NOT_ENOUGH_LOCK);
+
+        _transfer(_from, msg.sender, _to, _toLockManager, _amount);
+
+        // no need for SafeMath: checked in require above
+        lock.amount = lock.amount - _amount;
 
-        _transfer(_from, _fromLockId, _to, _toLockId, _amount);
-        _updateActiveLockAmount(_from, _fromLockId, _amount, false);
+        // update total
+        account.totalLocked = account.totalLocked.sub(_amount);
```

Comment:
Shouldn't we try to re-use `decreaseLockAmount` here?  we are missing the event otherwise

Reply:
Haha, it seems I removed the internal one, will get it back.